index/scorch/snapshot_index.go

```diff
@@ -430,3 +461,25 @@ func (i *IndexSnapshot) DocumentVisitFieldTerms(id index.IndexInternalID,
 
 	return nil
 }
+
+func stringsToMap(strsArr []string) map[string]bool {
+	if strsArr == nil {
+		return nil
+	}
+	strs := map[string]bool{}
```

Comment:
in go, when tracking existence of things in a map, it is recommend to use empty struct as a value instead of bool:

strs := map[string]struct{}{}

when checking, you do
if _, ok := strs[key]; ok {
  // key is in the map
}